core/eth/reader.go

```diff
@@ -509,10 +509,14 @@ func (t *Reader) GetOperatorStakesWithSocketForQuorums(ctx context.Context, quor
 		state[quorumID] = make(map[core.OperatorIndex]core.OperatorStakeWithSocket, len(result.Operators[i]))
 		for j, op := range result.Operators[i] {
 			operatorIndex := core.OperatorIndex(j)
+			operatorSocket, err := core.ParseOperatorSocket(result.Sockets[i][j])
+			if err != nil {
+				t.logger.Errorf("failed to parse operator socket from: %v, err: %v", result.Sockets[i][j], err)
```

Comment:
This error handling seems not right: Seeing a malformed socket is internal error (it shouldn't be accepted when registering the socket in the first place), and L519 still uses this socket and later returns the value to the caller.

Reply:
Agreed. I think this issue existed before. We were directly casting result.Sockets[i][j] into a string without any verification. This PR makes this a typed struct. 

Was trying to keep the focus of this PR to just the refactor while minimizing adding new behavior. Let me return the err here.